### DISCUSSION TITLE
Help with settings: mediaChannelEquals

### DISCUSSION THREAD

**Discussion by mtjoernelund** [punxaphil/custom-sonos-card#305] [Q&A] [answered]
As far as I can tell my use case is exactly why the override “mediaContentIdEquals” was added. When playing a radiostation the mediatitle changes to show the currently playing track. So essentially I get the right artwork when the stream is stopped or when no music is playing (as the media title at that time will be the name of the station for which I have added an override), but no artwork when a music track is playing because there is no match between the media title and the override I have created.

Therefore I think I need to use the override for “mediaContentIdEquals”. My problem is that I cannot wrap me head around where to get the information that I need to add here. I have tried to add the address of the stream that I am playing, but that did not work. I know that Sonos names the streams RINCON-something which the documentation also indicates that I should use, but I have no idea where to find this information. 

I have a sneaking suspicion that there is something stupidly simple that I am missing here… Please help…

**Comment by punxaphil** [comment]
http://homeassistant.local/developer-tools/state

**Reply by Levante75** [reply]
Thank you! 
Example :
mediaArtworkOverrides:
  - mediaContentIdEquals: x-rincon-mp3radio://https://icecast-qmusicbe-cdp.triple-it.nl/joe-gold.mp3
    imageUrl: /local/JoeGold.jpg

Status in developer tools:

group_members:
  - media_player.huiskamer
volume_level: 0.04
is_volume_muted: false
media_content_id: x-rincon-mp3radio://https://icecast-qmusicbe-cdp.triple-it.nl/joe-gold.mp3
media_content_type: music
media_title: Magic
media_channel: Joe Gold
shuffle: false
repeat: "off"

**Reply by Levante75** [reply]
Unfortunately no good solution!
The stream Sonos is using, is not always the same. So above solution works... --- just once!  
So could we have a setting :
mediaArtworkOverrides:
  - mediaChannelIdEquals

**Reply by punxaphil** [reply]
Let me know if you have another suggestion on how to solve it.

**Reply by Levante75** [reply]
Hi,
Is adding "media**Channel**IdEquals" in the sonos card hard to do? 
For the moment I added a "ifMissing: true" with a general selfmade artwork.
Anyway thanks for the good work, super solution, we do not use the Sonos App anymore!

**Reply by punxaphil** [reply]
Doesn't `mediaTitleEquals` work?

**Reply by Levante75** [reply]
mediaTitle changes with every track played. This is not what we meant. Just the artwork of the radio station to customize.

**Reply by punxaphil** [reply]
mediaChannelId. Is that available in http://homeassistant.local/developer-tools/state ?
Please share screendump

**Reply by Levante75** [reply]
Here it is
![media_player_state](https://github.com/punxaphil/custom-sonos-card/assets/63196199/369abcc2-96bf-451c-b086-92e4aa9e8f88)

**Comment by mtjoernelund** [comment]
Brilliantly simple. I knew that there was something obvious that I was missing. 

However, I still seem to have an issue. The artwork does not show up even after adding the media_content_id from the developer tools section. I’ve tried including the string both with and without double quotes. I have triple checked that I have copied the string over correctly and verified that the image url is correct and accessible. And I cannot seem to find anything on this in the log files. Pointers as to where I should start by looking?

**Comment by punxaphil** [comment, marked as answer]
Fixed in https://github.com/punxaphil/custom-sonos-card/releases/tag/v7.6.0. Use `mediaChannelEquals`
